includes/process-download.php

```diff
@@ -691,7 +691,21 @@ function edd_process_signed_download_url( $args ) {
 	}
 
 	$order_parts = explode( ':', rawurldecode( $_GET['eddfile'] ) );
-
+	
+	// Check to make sure not at download limit
+	if ( edd_is_file_at_download_limit( $order_parts[1], $order_parts[0], $order_parts[2], $order_parts[3] ) ) {
+		wp_die( apply_filters( 'edd_download_limit_reached_text', __( 'Sorry but you have hit your download limit for this file.', 'edd' ) ), __( 'Error', 'edd' ), array( 'response' => 403 ) );	
+	}
+	
+	// Make sure the link hasn't expired
+	$expire = $_GET['ttl'];
+	if ( base64_encode( base64_decode( $expire, true ) ) === $expire ) {
```

Comment:
We can't use base64 functions here. They can only be used in `deprecated-functions.php`.